src/sentry_plugins/splunk/plugin.py

```diff
@@ -0,0 +1,149 @@
+            label='Token',
+            secret=self.get_option('token', project),
+        )]
+
+    def get_host_for_splunk(self, event):
+        host = event.get_tag('server_name')
+        if host:
+            return host
+
+        user_interface = event.interfaces.get('sentry.interfaces.User')
+        if user_interface:
+            host = user_interface.ip_address
+            if host:
+                return host
+
+        return None
+
+    def get_event_payload(self, event):
+        props = {
+            'event_id': event.event_id,
+            'transaction': event.get_tag('transaction') or '',
+            'release': event.get_tag('sentry:release') or '',
+            'environment': event.get_tag('environment') or '',
+        }
+        if 'sentry.interfaces.Http' in event.interfaces:
+            http = event.interfaces['sentry.interfaces.Http']
+            headers = http.headers
+            if not isinstance(headers, dict):
+                headers = dict(headers or ())
+
+            props.update({
+                'request_url': http.url,
+                'request_method': http.method,
+                'request_referer': headers.get('Referer', ''),
+            })
+        if 'sentry.interfaces.Exception' in event.interfaces:
+            exc = event.interfaces['sentry.interfaces.Exception'].values[0]
+            props.update({
+                'exception_type': exc.type,
+                'exception_value': exc.value,
+            })
+        elif 'sentry.interfaces.Message' in event.interfaces:
+            msg = event.interfaces['sentry.interfaces.Message']
+            props.update({
+                'message': msg.formatted or msg.message,
+            })
+        return props
+
+    # http://dev.splunk.com/view/event-collector/SP-CAAAE6M
+    def post_process(self, event, **kwargs):
+        token = self.get_option('token', event.project)
+        index = self.get_option('index', event.project)
+        instance = self.get_option('instance', event.project)
+        if not (token and index and instance):
+            return
+
+        if not instance.endswith('/services/collector'):
+            instance = instance.rstrip('/') + '/services/collector'
+
+        source = self.get_option('source', event.project) or 'sentry'
+
+        rl_key = 'splunk:{}'.format(md5_text(token).hexdigest())
+        # limit splunk to 50 requests/second
+        if ratelimiter.is_limited(rl_key, limit=50, window=1):
+            return
+
+        payload = {
+            'time': int(event.datetime.strftime('%s')),
+            'source': source,
+            'index': index,
+            'event': self.get_event_payload(event),
+        }
+        host = self.get_host_for_splunk(event)
+        if host:
+            payload['host'] = host
+
+        session = http.build_session()
+        session.post(
+            instance,
+            json=payload,
+            # Splunk cloud instances certifcates dont play nicely
```

Comment:
I feel like this should be a configuration option imo. Seems not great if we're just never verifying SSL since we're sending sensitive data.

Reply:
I dont know if it will ever validate. If we make a configuration option, I have a feeling most people will just shoot themselves in the foot. Alternatively we have to support users uploading their own certificates.

Reply:
Is this the case with Splunk Cloud though? Or just for on-premise hosted Splunk? Does their SaaS spit out invalid certs?

Or is your comment referring to people running their own Splunk?

Reply:
the comment is correct - its splunk cloud thats not verifying with our default certs. On-premise splunk would be an entirely different problem.

Reply:
Oh. Weird. `¯\_(ツ)_/¯`